### PR TITLE
Remove IceGrid SessionTimeout properties

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR removes the 3 IceGrid SessionTimeout properties, and updates the doc-comment for the two public keepAlive operations. It also changes their implementation to be no-op.

### CODE REVIEW

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/SessionKeeper.java [hunk 1]
```diff
@@ -377,8 +372,6 @@ public void registerObservers() throws java.lang.Throwable {
     private final AdminSessionPrx _session;
     private final boolean _routed;
 
-    private java.util.concurrent.Future<?> _keepAliveFuture;
```
Comment (bernardnormier) [inline java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/SessionKeeper.java:380]: This was apparently a leftover from a previous PR. It wasn't set anywhere.

File: slice/IceGrid/Admin.ice [hunk 2]
```diff
@@ -753,9 +753,8 @@ module IceGrid
     /// @see Registry
     interface AdminSession extends Glacier2::Session
     {
-        /// Keep the session alive. Clients should call this operation regularly to prevent the server from reaping the
-        /// session.
-        /// @see Registry#getSessionTimeout
+        /// Keep the session alive. This operation is provided for backwards compatibility. As of Ice 3.8, there is no
```
Comment (bernardnormier) [inline slice/IceGrid/Admin.ice:756]: Keeping this operation and making it no-op is nicer for backwards compatibility.

File: cpp/src/IceGrid/SessionI.cpp [hunk 3]
```diff
@@ -62,24 +62,6 @@ BaseSessionI::BaseSessionI(const string& id, const string& prefix, const shared_
...
-void
-BaseSessionI::keepAlive(const Ice::Current& current)
-{
-    lock_guard lock(_mutex);
-    if (_destroyed)
-    {
-        throw Ice::ObjectNotExistException(__FILE__, __LINE__, current.id, "", "");
-    }
-
-    _timestamp = chrono::steady_clock::now();
```
Comment (bernardnormier) [inline cpp/src/IceGrid/SessionI.cpp:74]: We need to remove the code that reads _timestamp in a follow-up PR.

File: cpp/src/IceGrid/RegistryI.cpp [hunk 4]
```diff
@@ -250,7 +250,7 @@ RegistryI::startImpl()
     _master = _replicaName == "Master";
 
     // TODO: temporary. For now, synchronized with the default idle timeout.
-    _sessionTimeout = chrono::seconds(properties->getPropertyAsIntWithDefault("IceGrid.Registry.SessionTimeout", 60));
+    _sessionTimeout = chrono::seconds(60);
```
Comment (bernardnormier) [inline cpp/src/IceGrid/RegistryI.cpp:253]: To be removed in a follow-up PR.

File: slice/IceGrid/Admin.ice (hunk 2 above)
Comment (pepone) [reply slice/IceGrid/Admin.ice:756]: We should deprecate the operation.

**Review by pepone — APPROVED** [review]

File: cpp/src/IceGrid/AdminSessionI.h [hunk 5]
```diff
@@ -26,7 +26,7 @@ namespace IceGrid
 
         Ice::ObjectPrx _register(const std::shared_ptr<SessionServantManager>&, const Ice::ConnectionPtr&);
 
-        void keepAlive(const Ice::Current& current) override { BaseSessionI::keepAlive(current); }
+        void keepAlive(const Ice::Current&) override {} // no-op
```
Comment (externl) [inline cpp/src/IceGrid/AdminSessionI.h:29]: Maybe we should expand the comment to say why. Might not be obvious looking back at this in the future.

File: cpp/src/IceGrid/SessionI.h [hunk 6]
```diff
@@ -59,20 +57,20 @@ namespace IceGrid
 
         Ice::ObjectPrx _register(const std::shared_ptr<SessionServantManager>&, const Ice::ConnectionPtr&);
 
-        void keepAlive(const Ice::Current& current) override { BaseSessionI::keepAlive(current); }
+        void keepAlive(const Ice::Current&) final {} // no-op
```
Comment (externl) [inline cpp/src/IceGrid/SessionI.h:60]: Ditto.

**Review by externl — APPROVED** [review]

**Review by bentoi — APPROVED** [review]

File: slice/IceGrid/Admin.ice (hunk 2 above)
Comment (bernardnormier) [reply slice/IceGrid/Admin.ice:756]: I tried and it turns out the deprecated metadata is broken: #2085.

File: cpp/src/IceGrid/AdminSessionI.h (hunk 5 above)
Comment (bernardnormier) [reply cpp/src/IceGrid/AdminSessionI.h]: Fixed comment.

**Pull request merged**